server/asr-sd/whisper_diarization.py

```
@@ -0,0 +1,466 @@
+            except Exception as e:
+                print("\nError: Pyannote models require HuggingFace authentication.")
+                print("Please provide a HuggingFace token using --hf-token")
+                raise e
+
+        # Move diarization to the same device
+        if self.device == "cuda" and torch.cuda.is_available():
+            try:
+                self.diarization_pipeline.to(torch.device("cuda"))
+            except Exception:
+                # Not all components may support .to("cuda"); silently fall back
+                pass
```

Ruff (0.14.4)

`try`-`except`-`pass` detected, consider logging the exception

(S110)